src/services_models/services_constructor.jl

```diff
@@ -20,7 +20,20 @@ function construct_services!(
 )
     isempty(services_template) && return
     incompatible_device_types = get_incompatible_devices(devices_template)
-    for service_model in values(services_template)
+    # group service needs to be constructed last
```

Comment:
I admit this could be much nicer and more efficient, but I'm not sure how. Open for suggestions.

Reply:
an option would be to "leave" the group service to last. Is there only ever one service with `formulation == GroupReserve`? If so, something like this is an alternative:

```julia
    function _build!(service_model)
        @debug "Building $(service_model.service_type) with $(service_model.formulation) formulation"
        services = service_model.service_type[]
        if validate_services!(
            service_model.service_type,
            services,
            incompatible_device_types,
            sys,
        )
            construct_service!(
                psi_container,
                services,
                sys,
                service_model,
                devices_template,
                incompatible_device_types,
            )
        end
    end    
    groupservice = nothing
    for (key, service_model) in service_models
        if service_model === GroupReserve  # group service needs to be constructed last
            groupservice = key
            continue
        end
        build!(service_model)
    end
    groupservice === nothing || _build!(service_models[groupservice])

Reply:
There can be more than one service but currently not more than one template. I like this alternative! what do you guys think? 
From what I see, there are some `_build!` functions in PSI, Nick is this similarity intentional?

Reply:
cool :) 

> From what I see, there are some _build! functions in PSI, Nick is this similarity intentional?

No, i did not know about these functions. It seems that [one is superficially similar](https://github.com/NREL-SIIP/PowerSimulations.jl/blob/16e1b453ed21e714db58065e03601af4405b2e54/src/core/operations_problem.jl#L393) but does much more. `_build!` was just the first name that came to mind - feel free to go with something else.

Also, to be clear, the proposal is an internal function (actually a closure) _inside_ `construct_services!` - closing over e.g. `psi_container` among other things. It literally just fulls the contents of the `for`-loop out, to avoid duplicating the code. 

so the `construct_services!` function would be structured  like
```julia
funtion construct_services!(args...)
    ...
    function _build!(service_model)
       ...
    end
    for ....
        _build!(...)
    end
    _build!(...)
end
```

Reply:
Normally we don't use _build functions or device constructors. We have reserved those for building the simulations/operations structs. We try to keep the constructors pretty verbose to resemble code from AML's and make it easier power systems modelers to "read" the models.

Reply:
sorry, i'm not sure i understand the comment. to clarify, i'm not suggesting we use any different functionality than is already used, i'm just proposing a reorgnaising on some code inside this function

Reply:
Just implemented the proposed change. @jd-lara please let me know if that's ok.